exec/java-exec/src/main/java/org/apache/drill/exec/store/dfs/FileSelection.java

```diff
@@ -359,15 +363,30 @@ private static Path handleWildCard(final String root) {
     }
   }
 
-  private static String removeLeadingSlash(String path) {
-    if (path.charAt(0) == '/') {
+  public static String removeLeadingSlash(String path) {
+    if (!path.isEmpty() && path.charAt(0) == '/') {
       String newPath = path.substring(1);
       return removeLeadingSlash(newPath);
     } else {
       return path;
     }
   }
 
+  // Check if the path is a valid sub path under the parent after removing backpaths. Throw an exception if
+  // it is not
+  // We pass subpath in as a parameter only for the error message
+  public static boolean checkBackPaths(String parent, String combinedPath, String subpath) {
```

Comment:
This method can return void, since it never will return false. I see you use this during testing but it can be tested with void type as well.